kernel/sched.c

```diff
@@ -62,10 +62,12 @@ static inline int is_metairq(struct k_thread *thread)
 #endif
 }
 
+#if CONFIG_ASSERT
 static inline int _is_thread_dummy(struct k_thread *thread)
```

Comment:
Hm... clang doesn't admit "static inline" as a way to evade the "static function declared but unused" warning?  :(  That seems like a misfeature to me.  How else do you declare a function that may or may not be needed depending on kconfig, but not need to #if everything out?  Meh.  Not a big deal I guess.